### ISSUE TITLE
Make settings element IDs match their internal name

### ISSUE THREAD

**Issue by siikamiika** [FooSoft/yomichan#335]
The amount of options is quite large, so I think we need a convention for naming their `id` attributes. It would also enable automatically reading options from HTML with the help of a data attribute or two, but that's not the scope of this issue.

I think a naming scheme such as the following would make sense:
```javascript
options.scanning.enableSearchTags = $('#scanning.enable-search-tags').prop('checked');
```

To make it even easier to parse, it could be in `camelCase` just like in the object.

#296

**Comment by toasted-nutbread** [comment]
I already have something similar to this started, but the branch is out of date and will need some rebasing/refactoring at some point.

https://github.com/toasted-nutbread/yomichan/commits/settings-improvements

The branch has the original code for the JSON schemas which has already been merged, and then further builds upon it in a few ways:

* Commit "[Add data-option-target descriptors](https://github.com/toasted-nutbread/yomichan/commit/53859da069e37b9a9b4bab92be530e767f905469)" effectively does what you're talking about.
* Commit "[Update implementation of apiOptionsSet](https://github.com/toasted-nutbread/yomichan/commit/d871e020d30201f7aa1f08326f1ae8028c2d84e2)" extends the capabilities of `apiOptionsSet` so it can add/delete/etc more effectively. A longer term goal would be to make all options mutations performed via `apiOptionsSet`.
* Commit "[Create class to observe settings based on HTML](https://github.com/toasted-nutbread/yomichan/commit/d0afa69eca0c53d1c7da450735b3757bbd2c4722)" adds a class which observers DOM mutations on the settings page and automatically assigns settings observers to them using `apiOptionsSet`. This eliminates the need to write code.

**Comment by siikamiika** [comment]
Nice! I'll leave this issue open to remind about the existing branch.

**Comment by toasted-nutbread** [comment]
This issue has been addressed by #542 and subsequent settings refactoring PRs.
